core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1225,7 +1225,7 @@ private int acquireSubsetBatchRecords(
                     break;
                 }
 
-                if (offsetState.getValue().state != RecordState.AVAILABLE) {
+                if (offsetState.getValue().state != RecordState.AVAILABLE || offsetState.getValue().rollbackState() == RecordState.ACQUIRED) {
```

Comment:
I think that really it's the presence of a rollback state at all, and not whether it would roll back to ACQUIRED which is interesting. The test purpose of the complete test is to see whether the batch is unencumbered.